enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,342 @@
+
+It might be also interesting for a customer to show this new perspective
+only to users with specific permissions.
+
+Additional perspectives can be dynamically added with the 
+[Perspective extension](https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/extensions/perspectives.ts).
+
+### Goals
+
+- Allow cluster admins to hide the user perspectives.
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+- Validating the console resource in the operator and inform the cluster admins with an error because the operator
+doesn't know the available perspectives provided by the console plugins.
+- Allow cluster admins to customize the navigation menu along with perspectives for users or groups.
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective id and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── visibility
+|               ├── state
+|               ├── accessReview
+|                   ├── requiresAccessReview
+|                       ├── group
+|                       ├── resource
+|                       ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── visibility
+|               ├── state
+|               ├── accessReview
+|                   ├── requiresAccessReview
+|                   |   ├── group
+|                   |   ├── resource
+|                   |   ├── verb
+|                   ├── missingAccessReview
+|                       ├── group
+|                       ├── resource
+|                       ├── verb
+└── ...
+```
+
+Each perspective conforms to the following schema:
+
+- `id: string` , defines the id of perspective to disable. Incorrect or unknown ids will be ignored.
+
+- `visibility: object` , defines the state of perspective along with access review checks if needed for that perspective.
+
+- `state: Enabled/Disabled/AccessReview` , defines the perspective is enabled or disabled or access review check is required.
+
+// +optional
+- `accessReview: object`, defines required and missing access review checks.
+
+// +optional
+- `requiresAccessReview: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when all checks are successful.
+
+// +optional
+- `missingAccessReview: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when at least one check fails.
```

Comment:
Do we require that at least one of `requiresAccessReview` and `missingAccessReview` are configured? What would happen if neither were configured?

Reply:
Ignore the check and enable the perspective? This all needs to be documented (this is why I'm asking the questions)

Reply:
Added in the document.